packages/flame/lib/src/game/transform2d.dart

```diff
@@ -0,0 +1,203 @@
+import 'dart:math' as math;
+import 'package:flutter/foundation.dart';
+import 'package:vector_math/vector_math_64.dart';
+import 'notifying_vector2.dart';
+
+/// This class describes a generic 2D transform, which is a combination of
+/// translations, rotations, reflections and scaling. These transforms are
+/// combined into a single matrix, that can be either applied to a canvas,
+/// composed with another transform, or used directly to convert coordinates.
+///
+/// The transform can be visualized as 2 reference frames: a "global" and
+/// a "local". At first, these two reference frames coincide. Then, the
+/// following sequence of transforms is applied:
+///   - translation to point [position];
+///   - rotation by [angle] radians clockwise;
+///   - scaling in X and Y directions by [scale] factors;
+///   - final translation by [offset], in local coordinates.
+///
+/// The class is optimized for repeated use: the transform matrix is cached
+/// and then recalculated only when the underlying properties change. Moreover,
+/// recalculation of the transform is postponed until the matrix is actually
+/// requested by the user. Thus, modifying multiple properties at once does
+/// not incur the penalty of unnecessary recalculations.
+///
+/// This class implements the [ChangeNotifier] API, allowing you to subscribe
+/// for notifications whenever the transform matrix changes. In addition, you
+/// can subscribe to get notified when individual components of the transform
+/// change: [position], [scale], and [offset] (but not [angle]).
+///
+class Transform2D extends ChangeNotifier {
+  final Matrix4 _transformMatrix;
+  bool _recalculate;
+  double _angle;
+  final NotifyingVector2 _position;
+  final NotifyingVector2 _scale;
+  final NotifyingVector2 _offset;
+  static const tau = 2 * math.pi;
+
+  Transform2D()
+      : _transformMatrix = Matrix4.identity(),
+        _recalculate = true,
+        _angle = 0,
+        _position = NotifyingVector2(),
+        _scale = NotifyingVector2()..setValues(1, 1),
```

Comment:
```suggestion
        _scale = NotifyingVector2.all(1.0),
```
It should be possible to do it like this to skip the initial notification that would otherwise incur right?
Not that there are any listeners, but still.

Reply:
done